cmd/ci-reporter/cmd/github.go

```diff
@@ -0,0 +1,182 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/google/go-github/v34/github"
+)
+
+// This regex is getting used to identify sig lables on github issues
+var sigRegex = regexp.MustCompile(`sig/[a-zA-Z-]+`)
+
+var (
+	newColumn = GithubProjectBoardColumn{
+		ColumnTitle: "New/Not Yet Started",
+		ColumnID:    4212817,
+	}
+	underInvestigationColumn = GithubProjectBoardColumn{
+		ColumnTitle: "In flight",
+		ColumnID:    4212819,
+	}
+	observingColumn = GithubProjectBoardColumn{
+		ColumnTitle: "New/Not Yet Started",
+		ColumnID:    4212821,
+	}
+	resolvedColumn = GithubProjectBoardColumn{
+		ColumnTitle: "Resolved",
+		ColumnID:    6798858,
+	}
+)
+
+type (
+	// ColumnTitle title of a github project board column
+	ColumnTitle string
+	// ColumnID ID of a github project board column
+	ColumnID int64
+)
+
+// GithubProjectBoardColumn specifies a github project board column
+type GithubProjectBoardColumn struct {
+	ColumnTitle ColumnTitle
+	ColumnID    ColumnID
+}
+
+// GithubReportData defines the github report data structure
+type GithubReportData map[ColumnTitle][]IssueOverview
+
+// IssueOverview defines the data types of a github issue in github report data
+type IssueOverview struct {
+	// URL github issue url
+	URL string
+	// ID github issue id
+	ID int64
+	// Title github issue title
+	Title string
+	// Sigs kubernetes sigs that are referenced via label
+	Sigs []string
+}
+
+// GetGithubReportData used to request the raw report data from github
+func GetGithubReportData(cfg ReporterConfig) (GithubReportData, error) {
+	ciSignalProjectBoard := []GithubProjectBoardColumn{newColumn, underInvestigationColumn}
+
+	// if the short flag is not set observingColumn & resolvedColumn will be added to the report
+	if !cfg.ShortReport {
+		ciSignalProjectBoard = append(ciSignalProjectBoard, observingColumn, resolvedColumn)
+	}
+
+	githubReportData := map[ColumnTitle][]IssueOverview{}
+	for _, column := range ciSignalProjectBoard {
+		cards, err := getCardsFromColumn(cfg, column.ColumnID)
+		if err != nil {
+			return nil, err
+		}
+		githubReportData[column.ColumnTitle] = cards
+	}
+	return githubReportData, nil
+}
+
+// PrintGithubReportData used to print github report data out to the console
+func PrintGithubReportData(reportData map[ColumnTitle][]IssueOverview) {
+	for columnTitle, issuesInColumn := range reportData {
+		if len(issuesInColumn) > 0 {
```

Comment:
If a particular heading has no issues under it I feel we should still print its title for consistency in report outputs, WDYT?

Reply:
We likely change the format of the report in the future. (I was experimenting with a generic reporter form which abstracts the testrgid and github reporter, this may be the way to go - but not in this PR)